workflow-controller/controller-service/src/test/java/io/hyscale/controller/directive/BuildSpecUpdateTestHandler.java

```diff
@@ -18,50 +18,37 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
 
-import java.io.IOException;
-
-import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.stereotype.Component;
 
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 import io.hyscale.commons.exception.HyscaleException;
 import io.hyscale.commons.utils.WindowsUtil;
-import io.hyscale.controller.ControllerTestInitializer;
 import io.hyscale.controller.directive.impl.BuildSpecJsonHandler;
-import io.hyscale.controller.util.ServiceSpecTestUtil;
 import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
 import io.hyscale.servicespec.commons.model.service.Artifact;
 import io.hyscale.servicespec.commons.model.service.BuildSpec;
 import io.hyscale.servicespec.commons.model.service.ServiceSpec;
 
-@SpringJUnitConfig(classes = ControllerTestInitializer.class)
-public class BuildSpecJsonHandlerTest {
+@Component
+public class BuildSpecUpdateTestHandler implements ServiceSpecUpdateTestHandler {
+
+    private static final String buildSpecUpdateSpec = "/servicespecs/buildSpec_update.hspec.yaml";
 
     private static final String JSON_PATH = HyscaleSpecFields.getPath(HyscaleSpecFields.image,
             HyscaleSpecFields.buildSpec);
 
     @Autowired
     private BuildSpecJsonHandler buildSpecHandler;
 
-    private static ObjectNode serviceSpecNode = null;
-
-    private static ServiceSpec oldServiceSpec = null;
-
-    private static ServiceSpec updatedServiceSpec = null;
+    @Override
+    public String getServiceSpec() {
```

Comment:
This method can be refactored to a more relevant name, in fact it is not returning the service spec but the path to test resource of service spec